node_encoders.go

```diff
@@ -80,29 +80,40 @@ func encodeExtensionNode(pathBounds [2]byte, path, childData []byte) (data []byt
 
 // encodeSumInnerNode encodes an inner node for an smst given the data for both children
 func encodeSumInnerNode(leftData, rightData []byte) (data []byte) {
-	// Compute the sum of the current node
-	var sum [sumSizeBytes]byte
-	leftSum := parseSum(leftData)
-	rightSum := parseSum(rightData)
-	// TODO_CONSIDERATION: ` I chose BigEndian for readability but most computers
-	// now are optimized for LittleEndian encoding could be a micro optimization one day.`
-	binary.BigEndian.PutUint64(sum[:], leftSum+rightSum)
+	leftSum, leftCount := parseSumAndCount(leftData)
+	rightSum, rightCount := parseSumAndCount(rightData)
+
+	// Compute the SumBz of the current node
+	var SumBz [sumSizeBytes]byte
+	binary.BigEndian.PutUint64(SumBz[:], leftSum+rightSum)
+
+	// Compute the count of the current node
+	var countBz [countSizeBytes]byte
+	binary.BigEndian.PutUint64(countBz[:], leftCount+rightCount)
 
 	// Prepare and return the encoded inner node data
 	data = encodeInnerNode(leftData, rightData)
-	data = append(data, sum[:]...)
+	data = append(data, SumBz[:]...)
+	data = append(data, countBz[:]...)
 	return
 }
 
 // encodeSumExtensionNode encodes the data of a sum extension nodes
```

Comment:
```suggestion
// encodeSumExtensionNode encodes the data of a sum extension node
```
Unrelated but small detail